diff --git a/CPP/7zip/7zip_gcc.mak b/CPP/7zip/7zip_gcc.mak
index 5907421..1e78973 100644
--- a/CPP/7zip/7zip_gcc.mak
+++ b/CPP/7zip/7zip_gcc.mak
@@ -12,9 +12,11 @@ MY_ASM = asmc
 PROGPATH = $(O)/$(PROG)
 
 
+ifndef EMSCRIPTEN
 ifneq ($(CC), xlc)
 CFLAGS_WARN_WALL = -Wall -Werror -Wextra
 endif
+endif
 
 # for object file
 CFLAGS_BASE_LIST = -c
@@ -72,7 +74,11 @@ endif
 
 else
 
+ifdef EMSCRIPTEN
+LDFLAGS = $(LDFLAGS_STATIC) $(LDFLAGS_EMCC)
+else
 LDFLAGS = $(LDFLAGS_STATIC)
+endif
 # -s is not required for clang, do we need it for GGC ???
 # -s
 
diff --git a/CPP/7zip/Bundles/Alone2/makefile.emcc b/CPP/7zip/Bundles/Alone2/makefile.emcc
new file mode 100644
index 0000000..1e2361c
--- /dev/null
+++ b/CPP/7zip/Bundles/Alone2/makefile.emcc
@@ -0,0 +1,3 @@
+include ./makefile.gcc
+
+PROG = 7zz.js
diff --git a/CPP/7zip/Bundles/Alone2/post.js b/CPP/7zip/Bundles/Alone2/post.js
new file mode 100644
index 0000000..2ea3a96
--- /dev/null
+++ b/CPP/7zip/Bundles/Alone2/post.js
@@ -0,0 +1,3 @@
+Module.FS = FS;
+Module.NODEFS = NODEFS;
+Module.callMain = callMain;
\ No newline at end of file
diff --git a/CPP/7zip/Bundles/Alone2/pre.js b/CPP/7zip/Bundles/Alone2/pre.js
new file mode 100644
index 0000000..2084ccb
--- /dev/null
+++ b/CPP/7zip/Bundles/Alone2/pre.js
@@ -0,0 +1 @@
+Module.noInitialRun = true;
\ No newline at end of file
diff --git a/CPP/Windows/System.cpp b/CPP/Windows/System.cpp
index 099407e..2118c0a 100644
--- a/CPP/Windows/System.cpp
+++ b/CPP/Windows/System.cpp
@@ -6,6 +6,7 @@
 #include <unistd.h>
 #ifdef __APPLE__
 #include <sys/sysctl.h>
+#elif __EMSCRIPTEN__
 #else
 #include <sys/sysinfo.h>
 #endif
@@ -205,6 +206,8 @@ bool GetRamSize(UInt64 &size)
   
   // fixme
 
+  #elif defined(__EMSCRIPTEN__)
+
   #else
 
   struct sysinfo info;
